pool/app/session/repo/repo.ml

```diff
@@ -733,13 +733,27 @@ module Sql = struct
 
   let find_for_calendar_by_user actor pool ~start_time ~end_time =
     let open Caqti_request.Infix in
+    let open Utils.Lwt_result.Infix in
+    let experiment_checks = [ Format.asprintf "pool_experiments.uuid IN %s" ] in
+    let session_checks =
+      [ Format.asprintf "pool_sessions.uuid IN %s"
+      ; Format.asprintf "pool_sessions.experiment_uuid IN %s"
+      ]
+    in
+    let location_checks = [ Format.asprintf "pool_locations.uuid IN %s" ] in
     let%lwt guardian =
-      Guard.sql_where_fragment
-        ~field:"pool_experiments.uuid"
-        pool
-        Guard.Permission.Read
-        `Experiment
-        actor
+      [ session_checks, `Session, "pool_sessions.uuid IS NOT NULL"
+      ; experiment_checks, `Experiment, "pool_experiments.uuid IS NOT NULL"
+      ; location_checks, `Location, "pool_locations.uuid IS NOT NULL"
```

Comment:
Because of the combination of all those three, they will be in an "OR" statement. If you're allowed to read all e.g. sessions and experiments, it only adds the validation for location, which means, that you can only see the sessions for you least rights about the locations.

This is to make sure these filters are not "skipped".